### PR TITLE
Add support for Dropbox short-lived offline refresh tokens.

### PR DESCRIPTION
**Pull request by thunderkeys**
This patch adds support for offline refresh tokens for Dropbox, which will be required beginning September 2021.

https://dropbox.tech/developers/migrating-app-permissions-and-access-tokens

### CODE REVIEW

File: flask_dance/consumer/requests.py
Comment (singingwolfboy) [inline flask_dance/consumer/requests.py:148]: Modifying the `OAuth2Session` class seems like a larger change than supporting this change from Dropbox. Can you move it to a separate pull request? It's also going to need an automated test and updated documentation to support it.

**Review by singingwolfboy — CHANGES_REQUESTED** [review]
Hey @thunderkeys, thanks for the pull request! Sorry it took me so long to review it. :-(

We're going to need an automated test for this change before it can get merged. If you're not sure how to write one, I'm happy to help you with that if you want -- maybe we could find a time to videochat and paircode or something. :-) The change to `OAuth2Session` should also be its own pull request, rather than rolled into this one. Once you do those things, I think this can be merged!

Also, are you interested in being the maintainer for the Dropbox configuration? I don't actively use it these days, so it makes sense to have the maintainer be someone who _is_ using it.

**Comment by rooterkyberian** [comment]
@thunderkeys thanks for being vigilant about this - I would have missed if it wasn't for you.

Do you think you will find time to update your PR or can I recreate it addressing the reviewer comment so we can move forward and (hopefully) get it released?

In the mean time I'm using following workaround:
```
auth_blueprint = make_dropbox_blueprint(...)
auth_blueprint.authorization_url_params["token_access_type"] = "offline"
```

**Comment by thunderkeys** [comment]
Hi - I don't mind if you recreate it with the suggestions, it may be a bit before I have time to.  Good idea on the workaround.

File: flask_dance/consumer/requests.py
Comment (rooterkyberian) [reply flask_dance/consumer/requests.py:148]: this part of the PR is only thing not yet merged

rest was merged in some form in #339

**Comment by singingwolfboy** [comment]
Implemented by #339.

**Pull request closed without merging**
